x-pack/platform/plugins/shared/inference/server/chat_complete/adapters/bedrock/bedrock_claude_adapter.ts

```diff
@@ -51,12 +51,12 @@ export const bedrockClaudeAdapter: InferenceConnectorAdapter = {
       ...(metadata?.connectorTelemetry ? { telemetryMetadata: metadata.connectorTelemetry } : {}),
     };
 
-    return from(
-      executor.invoke({
+    return defer(() => {
+      return executor.invoke({
         subAction: 'invokeStream',
         subActionParams,
-      })
-    ).pipe(
+      });
+    }).pipe(
```

Comment:
Not directly required by the PR, but I changed the implementation for all adapters to use `defer` instead of `from`. In practice, that allows retrying / resetting the observables (e.g for retry on error mechanisms).

That required adapting the unit tests, as `defer` makes it lazily initiatize the observable on subscription, where `from` was doing that eagerly on execution.